lib/agent0/agent0/hyperdrive/exec/execute_agent_trades.py

```diff
@@ -39,28 +45,69 @@ async def async_execute_single_agent_trade(
         The HyperdriveAgent that is conducting the trade
     hyperdrive : HyperdriveInterface
         The Hyperdrive API interface object
+
+    Returns
+    -------
     """
     trades: list[types.Trade[HyperdriveMarketAction]] = agent.get_trades(interface=hyperdrive)
-    for trade_object in trades:
-        policy_name = agent.policy.__class__.__name__
-        logging.info(
-            "AGENT %s (%s) to perform %s for %g",
-            str(agent.checksum_address),
-            policy_name,
-            trade_object.market_action.action_type,
-            float(trade_object.market_action.trade_amount),
-        )
-        try:
-            wallet_deltas = await async_match_contract_call_to_trade(agent, hyperdrive, trade_object)
-            agent.wallet.update(wallet_deltas)
-        except UnknownBlockError as exc:
-            logging.error(exc)
+
+    # Make trades async for this agent. This way, an agent can submit multiple trades for a single block
```

Comment:
I don't think we need to explain how it used to work in a code comment, since that'll be less and less relevant as time goes on. Best to just explain how it works now.